testing/microlith-test/integration/prometheus_basic_auth/prometheus.bats

```diff
@@ -14,52 +14,64 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-load "$BATS_DETIK_ROOT/utils.bash"
-load "$BATS_DETIK_ROOT/linter.bash"
-load "$BATS_DETIK_ROOT/detik.bash"
 load "$BATS_SUPPORT_ROOT/load.bash"
 load "$BATS_ASSERT_ROOT/load.bash"
 load "$BATS_FILE_ROOT/load.bash"
 
-export SERVER_IMAGE=${SERVER_IMAGE:-couchbase/server:7.0.1}
-export CMOS_IMAGE=${CMOS_IMAGE:-couchbase/observability-stack:v1}
-export CMOS_PORT=${CMOS_PORT:-8080}
-
 setup() {
-    if [ "$TEST_INTEGRATION" == "true" ]; then
-        skip "Skipping integration tests"
+    if [ "$TEST_NATIVE" != "true" ]; then
+        skip "Skipping native prometheus tests"
     fi
 }
 
 teardown() {
-    if [ "$SKIP_TEARDOWN" == "true" ]; then
-        echo "# Skipping teardown. Make sure to manually run the commands in teardown()." >&3
-        return
+    if [ "$TEST_NATIVE" == "true" ]; then
+        docker-compose --project-directory="${BATS_TEST_DIRNAME}" logs --timestamps || echo "Unable to get compose output" >&3
+        docker-compose --project-directory="${BATS_TEST_DIRNAME}" rm -v --force --stop || true
     fi
-    docker-compose rm -v --force --stop
 }
 
-@test "Verify that basic auth can be passed by environment variable" {
-    docker-compose up -d --force-recreate --remove-orphans
-    # Wait for Couchbase to initialise
-    while true; do
-        if curl -s -o /dev/null -u Administrator:newpassword http://127.0.0.1:8091/pools/default; then
-            break
+waitForRemote() {
+    URL=$1
+    MAX_ATTEMPTS=$2
+    CREDENTIALS=$3
+    ATTEMPTS=0
+    until curl -s -o /dev/null "${CREDENTIALS}" "${URL}"; do
+        # shellcheck disable=SC2086
+        if [[ $ATTEMPTS -gt $MAX_ATTEMPTS ]]; then
+            assert_failure "unable to communicate with $URL"
         fi
+        ATTEMPTS=$((ATTEMPTS+1))
+        echo "Attempt $ATTEMPTS of $MAX_ATTEMPTS for $URL" >&3
+        sleep 10
     done
+    run curl -s "${CREDENTIALS}" "${URL}"
+    assert_success
+}
+
+@test "Verify pre-requisites" {
+    run : "${TEST_ROOT?"Need to set TEST_ROOT"}"
+    assert_success
+    run : "${CMOS_PORT?"Need to set CMOS_PORT"}"
+    assert_success
+}
```

Comment:
I'd move this to something at the top of the file, before any tests are run, otherwise we can't assume these variables are set in `setup()` which we may want to do.

Reply:
Yeah good shout. I think there is a canonical way to do this so you don't run any more tests within the file.